test/functional/apps/discover/_shared_links.js

```diff
@@ -60,47 +64,103 @@ export default function ({ getService, getPageObjects }) {
+            '/discover?_g=(refreshInterval:(pause:!t,value:0),time' +
+            ':(from:\'2015-09-19T06:31:44.000Z\',to:\'2015-09' +
+            '-23T18:31:44.000Z\'))&_a=(columns:!(_source),index:\'logstash-' +
+            '*\',interval:auto,query:(language:kuery,query:\'\')' +
+            ',sort:!(!(\'@timestamp\',desc)))';
+          const actualUrl = await PageObjects.share.getSharedUrl();
+          // strip the timestamp out of each URL
+          expect(actualUrl.replace(/_t=\d{13}/, '_t=TIMESTAMP')).to.be(
+            expectedUrl.replace(/_t=\d{13}/, '_t=TIMESTAMP')
+          );
+        });
+
+        it('should allow for copying the snapshot URL as a short URL', async function () {
+          const re = new RegExp(baseUrl + '/goto/[0-9a-f]{32}$');
+          await PageObjects.share.checkShortenUrl();
+          await retry.try(async () => {
+            const actualUrl = await PageObjects.share.getSharedUrl();
+            expect(actualUrl).to.match(re);
+          });
+        });
+
+        it('should allow for copying the saved object URL', async function () {
+          const expectedUrl =
+            baseUrl +
+            '/app/kibana#' +
+            '/discover/ab12e3c0-f231-11e6-9486-733b1ac9221a' +
+            '?_g=(refreshInterval%3A(pause%3A!t%2Cvalue%3A0)' +
+            '%2Ctime%3A(from%3A\'2015-09-19T06%3A31%3A44.000Z\'%2C' +
+            'to%3A\'2015-09-23T18%3A31%3A44.000Z\'))';
+          await PageObjects.discover.loadSavedSearch('A Saved Search');
+          await PageObjects.share.clickShareTopNavButton();
+          await PageObjects.share.exportAsSavedObject();
           const actualUrl = await PageObjects.share.getSharedUrl();
-          expect(actualUrl).to.match(re);
+          expect(actualUrl).to.be(expectedUrl);
         });
       });
+    });
+
+    describe('shared links with state in sessionStorage', async () => {
+      let teardown;
+      before(async function () {
+        teardown = await setup({ storeStateInSessionStorage: true });
+      });
 
-      it('should allow for copying the saved object URL', async function () {
-        const expectedUrl =
-          baseUrl +
-          '/app/kibana#' +
-          '/discover/ab12e3c0-f231-11e6-9486-733b1ac9221a' +
-          '?_g=(refreshInterval%3A(pause%3A!t%2Cvalue%3A0)' +
-          '%2Ctime%3A(from%3A\'2015-09-19T06%3A31%3A44.000Z\'%2C' +
-          'to%3A\'2015-09-23T18%3A31%3A44.000Z\'))';
-        await PageObjects.discover.loadSavedSearch('A Saved Search');
-        await PageObjects.share.clickShareTopNavButton();
-        await PageObjects.share.exportAsSavedObject();
-        const actualUrl = await PageObjects.share.getSharedUrl();
-        expect(actualUrl).to.be(expectedUrl);
+      after(async function () {
+        await teardown();
+      });
+
+      describe('permalink', function () {
+        it('should allow for copying the snapshot URL as a short URL and should open it', async function () {
```

Comment:
This is the new tests I've added. 
1. It sets state:storeInSessionStorage flag
2. generates shorten url
3. cleans up sessionStorage and actually opens shorten url and checks that discover is loaded and time range filter is preserved